### PR TITLE
Access control: Load permissions from memory and database

### PR DESCRIPTION
**Pull request by kalleep**
Now that we have the access control database in oss we need to load permissions from both in memory structures (fixed roles) and from the database.

This pr update GetUserPermissions to load permissions from in memory structure and from the database.

### CODE REVIEW

File: pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go [hunk 1]
```diff
@@ -82,28 +84,49 @@ func (ac *OSSAccessControlService) GetUserPermissions(ctx context.Context, user
 	timer := prometheus.NewTimer(metrics.MAccessPermissionsSummary)
 	defer timer.ObserveDuration()
 
-	builtinRoles := ac.GetUserBuiltInRoles(user)
+	permissions := ac.getFixedPermissions(ctx, user)
```
Comment (gamab) [inline pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go:87]: Nit. you could provide the user's built-in role to call `ac.GetUserBuiltInRoles` once. (however that's not an expensive call)

**Review by gamab — APPROVED** [review]
LGTM!

What do you think about adding a test to `GetUserPermissions` with permissions coming from a mock `PermissionsProvider`?

I saw you had a corresponding branch on ENT. Do you want to open a PR for review?

**Review by sakjur — CHANGES_REQUESTED** [review]
Talked with @kalleep yesterday, let's wait with this for a bit 🙂

File: pkg/api/common_test.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/grafana/grafana/pkg/infra/usagestats"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	acdb "github.com/grafana/grafana/pkg/services/accesscontrol/database"
```
Comment (IevaVasiljeva) [inline pkg/api/common_test.go:23]: nit: we can maybe call this `accesscontroldb` instead to make it a bit more readable and consistent with how we import `accesscontrolmock` dependency?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM, but I couldn't make it run due to wire failing to initialize dependencies (I think permission provider is required before it's added). But I might have done something wrong there.

File: pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go [hunk 3]
```diff
@@ -138,7 +164,7 @@ func (ac *OSSAccessControlService) saveFixedRole(role accesscontrol.RoleDTO) {
 		// needs to be increased. Hence, we don't overwrite a role with a
 		// greater version.
 		if storedRole.Version >= role.Version {
-			ac.Log.Debug("the has already been stored in a greater version, skipping registration", "role", role.Name)
+			ac.log.Debug("the has already been stored in a greater version, skipping registration", "role", role.Name)
```
Comment (gamab) [inline pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go:167]:
```suggestion
			ac.log.Debug("the role has already been stored in a greater version, skipping registration", "role", role.Name)
```

**Review by gamab — APPROVED** [review]
LGTM

**Review by sakjur — APPROVED** [review]
I'm happy with the changes here :+1:

**Pull request merged**